### PR TITLE
CDF-15369 Return better error for wrongly typed data model property values

### PR DESCRIPTION
**Pull request by emelvarol-cognite**
…model

### CODE REVIEW

File: src/test/scala/cognite/spark/v1/DataModelInstancesRelationTest.scala [hunk 1]
```diff
@@ -198,6 +198,22 @@ class DataModelInstancesRelationTest
...
+        spark
+          .sql(s"""select '2.0' as prop_float,
+                  |true as prop_bool,
+                  |'abc' as prop_string,
+                  |'test' as externalId""".stripMargin)
+      )
+    }
+    ex shouldBe an[CdfSparkException]
+    ex.getMessage shouldBe "2.0 of type class java.lang.String is not a valid float64. " +
+      "Try to cast the value to double. For example, ‘double(col_name) as prop_name’ or ‘cast(col_name as double) as prop_name’."
```
Comment (vuhainguyen) [inline src/test/scala/cognite/spark/v1/DataModelInstancesRelationTest.scala:214]: Maybe you can also add 2 more tests with query : `select cast('2.0' as double) as prop_float` and `select double('2.0')  as prop_float`to make sure the casting work 😁

Comment (emelvarol-cognite) [reply src/test/scala/cognite/spark/v1/DataModelInstancesRelationTest.scala:214]: aha good idea, let me do it

Comment (emelvarol-cognite) [reply src/test/scala/cognite/spark/v1/DataModelInstancesRelationTest.scala:216]: can you check?

**Review by vuhainguyen — APPROVED** [review]

File: src/test/scala/cognite/spark/v1/DataModelInstancesRelationTest.scala (hunk 1 above)
Comment (vuhainguyen) [reply src/test/scala/cognite/spark/v1/DataModelInstancesRelationTest.scala:216]: 😸 ✅

**Pull request merged**